src/beast/beast/nudb/detail/bulkio.h

```diff
@@ -45,10 +45,16 @@ class bulk_reader
     bulk_reader (File& f, std::size_t offset,
         std::size_t last, std::size_t buffer_size);
 
+    std::size_t
+    offset() const
+    {
+        return offset_ - avail_;
```

Comment:
`offset_` is where the bulk reader is located, not what will be returned to the caller. `offset_ - avail` is where the caller things the next byte will come from. This code is correct as written.